src/main/java/jfdi/parser/InputParser.java

```diff
@@ -0,0 +1,69 @@
+package jfdi.parser;
+
+import java.util.Arrays;
+import java.util.List;
+
+import jfdi.logic.interfaces.AbstractCommand;
+import jfdi.parser.commandparsers.AddCommandParser;
+import jfdi.parser.commandparsers.DeleteCommandParser;
+import jfdi.parser.commandparsers.ListCommandParser;
+
+/**
+ * The Parser class exposes the key API for the parser component.
+ *
+ * @author leona_000
+ *
+ */
+public class InputParser implements IParser {
+    private static InputParser parserInstance;
+
+    public static InputParser getInstance() {
```

Comment:
Just wondering, should we overwrite the default constructor and make it private so that it would not be initialized unknowingly?